src/components/header/Header.jsx

```diff
@@ -1,34 +1,133 @@
+          <div className="logo">
+            <a href="../../referenceapplication/home.page">
+              <img 
+                alt="" 
+                src="../../ms/uiframework/resource/uicommons/images/logo/openmrs-with-title-small.png"
+              />
+            </a>
+          </div>
+
+          <ul className="user-options">
+            <li
+              className="identifier"
+              onClick={() => { this.toggleState("userDropdown"); }}
+              role="button"
+              style={{ cursor: "pointer" }}
+              tabIndex="0"
+            >
+              <i className="icon-user small" />
+              {this.props.currentUser}
+
+              <i className="icon-caret-down appui-icon-caret-down link" />
+              <i 
+                className="icon-caret-up link appui-toggle"
+                style={{ display: "none" }}
+              />
+              {this.state.userDropdown &&
+                <ul id="user-account-menu">
+                  <li>
+                    <a
+                      href="../../adminui/myaccount/myAccount.page"
+                    >Account
+                    </a>
+                  </li>
+                </ul>
+              }
+
+            </li>
+            <li className="change-location">
+              <a 
+                href="#"
+                onClick={() => { this.toggleState("locationDropdown"); }}
+              >
+                <i className="icon-map-marker small" />
+                <span id="selected-location">{this.props.currentLocation.display}</span>
+
+                <i className="link icon-caret-down" />
+              </a>
+              {this.state.locationDropdown &&
+                <div>
+                  <ul className="location-container">
+                    {this.props.locations.map(location => (
+                      <li
+                        className={location.display === this.props.currentLocation.display ? "selected" : ""}
+                        key={location.uuid}
+                        onClick={() => {
+                          this.props.dispatch(sessionActions.setSession(location.uuid));
+                          this.toggleState("locationDropdown", false);
+                        }}
+                        role="button"
+                      >
+                        {location.display}
+                      </li>
+                    ))}
+                  </ul>
+                </div>
+              }
+            </li>
+            <li className="logout">
+              <a href={`../../appui/header/logout.action?successUrl=${contextPath}`}>
+                Logout
+                <i className="icon-signout small" />
+              </a>
+            </li>
+          </ul>
         </header>
       </div>
     );
-  };
+  }
 }
 
+const mapStateToProps = (state) => {
+  const { currentLocation, currentUser } = state.openmrs.session;
+  const { locationTags } = state.openmrs.loginLocations;
```

Comment:
I would hope that state.openmrs.loginLocations would just be a list of login locations, so you could just go:

`const locations = state.openmrs.loginLocations`

Is it not this way?  If so, we should consider changing this code (I know it's not code you wrote).

Reply:
Fixed.